### DISCUSSION TITLE
Using realtime api what exactly is my ClientId

### DISCUSSION THREAD

**Discussion by Eko450-eng** [pocketbase/pocketbase#1940] [Q&A] [answered]
I basically wanted to try out the /api/realtime SSE but I seem to be not understanding what clientId is expected for the post request

Head contains: "Authorization": pb.authStore.token
And body contains: "clientId": pb.authStore.model.id

The user is logged in

```javascript
 const data = await fetch(`${process.env.NEXT_PUBLIC_DBURL}/api/realtime`, {
    method: "POST",
    mode: "cors",
    headers: head,
    body: JSON.stringify(bodyData)
  })
  const returnValue = await data.json()
  console.log("Post", returnValue)
  return returnValue
```

All i get is { code: 404, message: 'Missing or invalid client id.', data: {} }

**Comment by ganigeorgiev** [comment, marked as answer]
`clientId` is the unique client id that is returned as part of the `PB_CONNECT` message of the actual SSE connection (aka. `GET /api/realtime`).

The `POST /api/realtime` action "links" the SSE client with the submitted subscriptions.

You'll receive the event messages in the long-lived SSE connection (aka. as streamed body of `GET /api/realtime`).

I'm not sure what is your use case, but I'd recommend to use the [JS SDK](https://github.com/pocketbase/js-sdk) for subscribing to the realtime events:

```js
await pb.collection("example").subscribe("*", (e) => {
    console.log(e)
})
```

**Reply by Eko450-eng** [reply]
That works too, it was literrally just me trying to try out using it without the sdk :D thanks
